include/net/socket.h

```diff
@@ -409,6 +409,18 @@ static inline char *inet_ntop(sa_family_t family, const void *src, char *dst,
 #define EAI_SERVICE DNS_EAI_SERVICE
 #endif /* defined(CONFIG_NET_SOCKETS_POSIX_NAMES) */
 
+#define SOL_SOCKET 1
+
+/* Socket options for SOL_SOCKET level */
+#define SO_REUSEADDR 2
+#define SO_ERROR 4
+
+/* Socket options for IPPROTO_TCP level */
+#define TCP_NODELAY 1
+
+/* Socket options for IPPROTO_IPV6 level */
+#define IPV6_V6ONLY 26
```

Comment:
Is this needed? You do not have check for this value in `zsock_setsockopt_ctx()`

Reply:
Yes, everything in this patch is needed to build, then run open62541. All symbols are required to build it, but default runtime config of course doesn't expose all the codepaths, e.g. default is IPv4 address, which doesn't exercise IPv6 options.

Reply:
Ok